.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,58 @@
+name: Emscripten
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  js-build:
+    name: Build liblouis for js
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    # Contains tests and js snippets appended to builds. liblouis-js
+    # version should be incremented by hand, to keep the repositories
+    # in sync.
+    - name: Obtain liblouis-js
+      run: git clone https://github.com/liblouis/liblouis-js.git
+
+    - name: Pull the docker image
+      run: docker pull dolp/liblouis-js-build-travis:1.37.3-64bit
+
+    - name: Run the docker image
+      run: docker run --rm -v $(pwd):/src dolp/liblouis-js-build-travis:1.37.3-64bit /bin/bash  ".travis/script/emscripten-build.sh"
+
+    - name: Move the build result to a better place
+      run: |
+        cp -r ./out/* ./liblouis-js/js-build
+        cp -r ./out-emscripten-install/share/liblouis/tables ./liblouis-js/js-build/
+
+    - name: Echo phantomjs version
+      run: phantomjs --version
+
+    # `npm link` exposes the newly build npm package to the test
+    # runner. The `--production`-flag is necessary as it ensures that
+    # no publicly published package of the build is downloaded and
+    # tested instead.
+    - name: Build and test the npm package
+      run: |
+        cd ./liblouis-js
+        # FIXME: unclear why the npm link is needed
+        sudo npm link js-build --production
+        npm test
+        cd ..
+
+    - name: Upload Release Asset
+      id: upload-release-asset
```

Comment:
What are ids for?

Reply:
The just give an id to a step. I guess that is useful if you want to refer to results of a step later. But in our case it isn't needed.

Reply:
Yes I realized after I asked the question. The example from which this was copied had steps that were linked.